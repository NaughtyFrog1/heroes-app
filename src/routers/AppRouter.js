// Por convención se llama AppRouter para indicar que es el router principal de la aplicación

import React, { useContext } from 'react'
import { BrowserRouter as Router, Switch } from 'react-router-dom'
import AuthContext from '../auth/AuthContext'
import LoginScreen from '../components/login/LoginScreen'
import DashboardRoutes from './DashboardRoutes'
import PrivateRoute from './PrivateRoute'
import PublicRoute from './PublicRoute'

const AppRouter = () => {
  const { user } = useContext(AuthContext)

  return (
    <Router>
      <div>
        <Switch>
          <PublicRoute
            exact
            path="/login"
            component={LoginScreen}
            isAuthenticated={user.logged}
          />

          {/* 1. */}
          <PrivateRoute
            path="/"
            component={DashboardRoutes}
            isAuthenticated={user.logged}
          />
        </Switch>
      </div>
    </Router>
  )
}

export default AppRouter

/**
 * 1. Como en ese router estan todas las rutas que queremos proteger, solo debemos proteger ese componente.
 */
