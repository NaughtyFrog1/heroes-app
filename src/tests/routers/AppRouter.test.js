import { mount } from 'enzyme'
import AuthContext from '../../auth/AuthContext'
import AppRouter from '../../routers/AppRouter'

describe('tests in AppRouter', () => {
  test('should render login if not logged in', () => {
    const contextValue = {
      dispatch: jest.fn(),
      user: {
        logged: false,
      },
    }

    // Como el <AppRouter /> est√° dentro de un high order component debemos utilizar `mount` en lugar de `shallow`
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    )

    expect(wrapper).toMatchSnapshot()
  })

  test('should render marvel component if authenticated', () => {
    const contextValue = {
      dispatch: jest.fn(),
      user: {
        logged: true,
        name: 'John Doe',
      },
    }

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    )

    expect(wrapper.find('.navbar').exists()).toBe(true)
  })
})
