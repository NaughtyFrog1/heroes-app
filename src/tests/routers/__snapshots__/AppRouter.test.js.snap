// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in AppRouter should render login if not logged in 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login Screen
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
